engine/access/state_stream/backend/handler.go

```diff
@@ -70,7 +72,10 @@ func (h *Handler) GetExecutionDataByBlockID(ctx context.Context, request *execut
 		return nil, status.Errorf(codes.Internal, "could not convert execution data event payloads to JSON: %v", err)
 	}
 
-	return &executiondata.GetExecutionDataByBlockIDResponse{BlockExecutionData: message}, nil
+	return &executiondata.GetExecutionDataByBlockIDResponse{
+		BlockExecutionData: message,
+		ExecutorMetadata:   convert.ExecutorMetadataToMessage(executorMetadata),
```

Comment:
this should only include the metadata if `query.GetIncludeExecutorMetadata()` is true